workers/unity/Packages/io.improbable.gdk.buildsystem/Configuration/BuildEnvironmentConfig.cs

```diff
@@ -43,5 +45,45 @@ public BuildEnvironmentConfig(IEnumerable<BuildTarget> availableTargets, params
                 }
             }
         }
+
+        public IEnumerable<BuildTargetConfig> GetSupportedTargets(string workerType, BuildContextFilter contextFilter)
+        {
+            // Filter targets for CI
+            var targetConfigs = BuildTargets
+                .Where(t => t.Enabled && (contextFilter.BuildTargetFilter?.Contains(t.Target) ?? true))
+                .ToList();
+
+            // Filter out any deprecated targets
+            var supportedTargets = targetConfigs
+                .Where(c => !c.Deprecated)
+                .ToList();
+
+            // Which build targets are not supported by current install?
+            var missingTargets = supportedTargets
+                .Where(c => !BuildSupportChecker.CanBuildTarget(c.Target))
+                .ToList();
+
+            // Error on missing required build support
+            if (missingTargets.Any(c => c.Required))
+            {
+                var targetNames = string.Join(", ", missingTargets
+                    .Where(c => c.Required)
+                    .Select(c => c.Target.ToString()));
+                throw new BuildFailedException(
+                    $"Build failed for {workerType}. Cannot build for required ({targetNames}) because build support is not installed in the Unity Editor.");
```

Comment:
Seems a bit like a code smell that we pass in the `workerType` just to add context to this exception. 

The caller could equally add this context right?